edx_rest_api_client/client.py

```diff
@@ -97,6 +99,10 @@ def __init__(self, url, signing_key=None, username=None, full_name=None, email=N
         session = session or requests.Session()
         session.headers['User-Agent'] = self.user_agent()
 
+        if retry_attempts > 0:
+            retries = Retry(total=retry_attempts, backoff_factor=2)
```

Comment:
There are a lot of options on Retry.  Not sure that we want to retry for all status codes (like 403, 404, etc.?).
I know other code used the `@backoff` decorator.  I guess I'd get @feanil to weigh-in on the implementation.

Also, this needs some documentation somewhere.  
- I'd also include some best practices in there around situations where retry is ok and where it is not, and one might take the timeout into consideration when they really want to retry.

Reply:
Try has a set of default status codes it will retry on. Do we want to add that as a note here? 

Added some notes to the retry section